1.data-exploration/scripts/nbconverted/sample_info_data_visualization.py

```diff
@@ -0,0 +1,180 @@
+df_gene_dependency.head(1)
+
+
+# In[5]:
+
+
+# check if "figures" directory exists and create directory if it does not
+fig_dir = pathlib.Path("figures")
+fig_dir.mkdir(exist_ok=True)
+
+
+# In[6]:
+
+
+# sample_info.csv visualization
+# how many samples from sample_info.csv?
+n_samples = len(df_sample_info["DepMap_ID"].unique())
+print(f"Number of Samples Documented in sample_info.csv: {n_samples} \n")
+
+# how many samples from CRISPR_gene_dependency.csv?
+n_samples2 = len(df_gene_dependency["DepMap_ID"].unique())
+print(f"Number of Samples Included in CRISPR_gene_dependency.csv: {n_samples2} \n")
+
+# how many different ages were sampled from? 
+all_ages = df_sample_info["age"].unique()
+print(f"Ages sampled from: \n {all_ages} \n")
+
+
+# how many different types of cancer?
+all_cancers = df_sample_info["primary_disease"].unique()
+print(f"All Cancer Types: \n {all_cancers} \n")
+
+# create a bar chart that shows the number of types of cancer sampled 
+data = df_sample_info
+cancer_types_bar = (
+    gg.ggplot(data, gg.aes(x="primary_disease")) + gg.geom_bar() + gg.theme(axis_text_x =element_text(angle = 90))
+    )
+print(cancer_types_bar)
+cancer_types_bar.save("./figures/sample_cancer_types_bar_chart.png")
+
+# identify which samples are included in both sample_info.csv and CRISPR_gene_dependency.csv
+similar_samples = list(set(df_sample_info["DepMap_ID"]) & set(df_gene_dependency["DepMap_ID"]))
+
+# count the number of samples that overlap in both data sets 
+sample_overlap = len(similar_samples)
+print(f"number of sample overlaps between sample_info.csv and CRISPR_gene_dependency.csv: {sample_overlap} \n")
+
+
+# In[7]:
+
+
+age_vector_to_clean = df_sample_info.loc[:, "age"].tolist()
+
+age_categories = []
+age_distribution = []
+
+adult_threshold = 18
+
+# Loop through each age entry to clean it
+for age_entry in age_vector_to_clean:
+    try:
+        # If the age is an integer, apply appropriate category
+        if int(age_entry) >= adult_threshold:
+            age_categories.append("Adult")
+        else:
+            age_categories.append("Pediatric")
+        
+        # If the age is an integer, apply appropriate continuous measure
+        age_distribution.append(int(age_entry))
+
+    except ValueError:
+        # If conversion fails, categorize appropriately
+        if pd.notnull(age_entry):
+            age_categories.append(age_entry)
+        else:
+            age_categories.append("Missing")
+        
+        age_distribution.append(np.nan)
+
+
+# In[8]:
+
+
+# New dataframe containing two new columns age_categories & age_distribution
+df_age_visual = (
+    df_sample_info.assign(
+        age_categories=age_categories,
+        age_distribution=age_distribution
+    )
+)
+
+df_age_visual.head()
+
+
+# In[9]:
+
+
+age_categories_bar = (
+    gg.ggplot(df_age_visual, gg.aes(x="age_categories"))
+    + gg.geom_bar()
+)
+print(age_categories_bar)
+age_categories_bar.save("./figures/age_categories_bar_chart.png")
```

Comment:
```suggestion
bar_output_file = pathlib.Path("./figures/age_categories_bar_chart.png")
age_categories_bar.save(bar_output_file)
```

Make sure all paths are predefined using pathlib.